benchbuild/utils/actions.py

```diff
@@ -688,3 +659,34 @@ def __str__(self, indent: int = 0) -> str:
                 project.name, version_str
             ), indent * " "
         )
+
+
+class SetProjectVersion(Step):
+    NAME = 'SET PROJECT VERSION'
+    DESCRIPTION = 'Checkout a project version'
+
+    def __init__(self, project: tp.Any, variant: source.Variant) -> None:
```

Comment:
`variant` is hard to come by from a user perspective, is it possible to just provide the project revision hash here?

```suggestion
    def __init__(self, project: tp.Any, version: CommitHash) -> None:
```